tools/tkn-results/internal/client/client.go

```diff
@@ -35,6 +35,13 @@ func NewDefaultFactory() (*ClientFactory, error) {
 	if err != nil {
 		return nil, err
 	}
+	if cfg.ServiceAccount.Name != "" && cfg.ServiceAccount.Namespace == "" {
```

Comment:
If I don't pass any argument and neither provide one in the config, it gives segmentation fault. It should rather exit with a defined error.

```
./tkn-results list default     
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x15e0232]

goroutine 1 [running]:
github.com/tektoncd/results/tools/tkn-results/internal/client.NewDefaultFactory()
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/internal/client/client.go:38 +0xd2
github.com/tektoncd/results/tools/tkn-results/internal/client.DefaultClient({0x1c34770, 0xc000188000}, {0x0, 0x0})
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/internal/client/client.go:91 +0x2d
github.com/tektoncd/results/tools/tkn-results/cmd.Root.func1(0xc000004c00?, {0xc0001f4a40?, 0x1?, 0x1?})
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/cmd/root.go:54 +0x165
github.com/spf13/cobra.(*Command).execute(0xc000004c00, {0xc0001f4a10, 0x1, 0x1})
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:891 +0x739
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004900)
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/spf13/cobra.(*Command).ExecuteContext(0x0?, {0x1c34770?, 0xc000188000?})
        /home/avinal/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:961 +0x4a
main.main()
        /home/avinal/Documents/redhat/github.com/tektoncd/results/tools/tkn-results/main.go:9 +0x2c
```

Reply:
It should have been picked from the kubeconfig. Strange. Let me check.

Reply:
This is fixed but another issue has been observed in main.